test/middleware.js

```diff
@@ -0,0 +1,212 @@
+        expect(defs).to.have.length(7);
+        var middleware = defs.filter(function(m) {
+          return !m.isPhasePlaceHolder;
+        });
+        expect(middleware).to.have.length(4);
+        done();
+      });
+    });
+    describe('config file', function() {
+      it('should be created', function(done) {
+        this.configFile.exists(function(err, exists) {
+          expect(err).to.not.exist;
+          expect(exists).to.equal(true);
+          done();
+        });
+      });
+      it('should not contain id properties', function() {
+        var configData = this.configFile.data;
+        var dsConfig = configData.routes.foo;
+        expect(dsConfig).to.not.have.property('id');
+        expect(dsConfig).to.not.have.property('facetName');
+      });
+      it('should contain phase place holder', function() {
+        var configData = this.configFile.data;
+        expect(configData.myPhase).exist;
+        expect(Object.keys(configData.myPhase)).to.eql([]);
+      });
+    });
+
+    it('should keep the order of entries', function(done) {
+      var defs = this.configFile.data;
+      expect(Object.keys(defs)).to.eql(
+        ['auth', 'routes:before', 'routes', 'myPhase']);
+      var routes = defs.routes;
+      expect(Object.keys(routes)).to.eql(['foo', 'bar']);
+      done();
+    });
+
+    it('should not contain workspace-private properties', function(done) {
+      var configFile = this.configFile;
+      Middleware.create({
+        name: 'another-middleware',
+        params: {x: 'rest'},
+        facetName: this.serverFacet
+      }, function(err) {
+        if (err) return done(err);
+        configFile.load(function(err) {
+          if (err) done(err);
+          var middlewares = configFile.data;
+          expect(Object.keys(middlewares.routes.foo)).to.not.contain('configFile');
+          done();
+        });
+      });
+    });
+
+    it('should add phase after a given phase', function(done) {
+      var configFile = this.configFile;
+      Middleware.addPhase(this.serverFacet, 'phase1', 'routes', function(err) {
+        if (err) return done(err);
+        configFile.load(function(err) {
+          if (err) done(err);
+          var middlewares = configFile.data;
+          expect(Object.keys(middlewares)).to.eql(
+            ['auth', 'phase1', 'routes:before', 'routes', 'myPhase']);
+          done();
+        });
+      });
+    });
+
+    it('should not add a phase if it exists', function(done) {
+      var configFile = this.configFile;
+      Middleware.addPhase(this.serverFacet, 'myPhase', 'routes', function(err) {
+        if (err) return done(err);
+        configFile.load(function(err) {
+          if (err) done(err);
+          var middlewares = configFile.data;
+          expect(Object.keys(middlewares)).to.eql(
+            ['auth', 'routes:before', 'routes', 'myPhase']);
+          done();
+        });
+      });
+    });
+
+    it('should add phase after the first phase', function(done) {
```

Comment:
I think this adds the new phase at the end, i.e. the test should be called "...after the last phase".